engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/command/ExecuteExpressionCommand.java

```diff
@@ -0,0 +1,40 @@
+package org.enso.interpreter.instrument.command;
+
+import java.util.UUID;
+import org.enso.interpreter.instrument.execution.RuntimeContext;
+import org.enso.interpreter.instrument.job.ExecuteExpressionJob;
+import org.enso.interpreter.instrument.job.ExecuteJob;
+import org.enso.polyglot.runtime.Runtime$Api$VisualizationAttached;
+import scala.Option;
+import scala.concurrent.ExecutionContext;
+import scala.concurrent.Future;
+import scala.runtime.BoxedUnit;
+
+public final class ExecuteExpressionCommand extends ContextCmd {
+
+  private final UUID contextId;
+  private final UUID visualizationId;
+  private final UUID expressionId;
+  private final String expression;
+
+  public ExecuteExpressionCommand(
+      Option<UUID> maybeRequestId,
+      UUID contextId,
+      UUID visualizationId,
+      UUID expressionId,
+      String expression) {
+    super(contextId, maybeRequestId);
+    this.contextId = contextId;
+    this.visualizationId = visualizationId;
+    this.expressionId = expressionId;
+    this.expression = expression;
+  }
+
+  @Override
+  public Future<BoxedUnit> executeCmd(RuntimeContext ctx, ExecutionContext ec) {
+    reply(new Runtime$Api$VisualizationAttached(), ctx);
+    return ctx.jobProcessor()
+        .run(new ExecuteExpressionJob(contextId, visualizationId, expressionId, expression))
+        .flatMap(executable -> ctx.jobProcessor().run(ExecuteJob.apply(executable)), ec);
```

Comment:
Is this the place that starts `Executable`?

Reply:
Yes, the visualization job is implemented the same way. We set up the state, and then run the program to capture the necessary node